source/module_base/lapack_connector.h

```diff
@@ -159,6 +159,20 @@ class LapackConnector
         return aux;
     }
 
+    static inline
+    std::complex<double>* transpose(const std::complex<double>* a, const int n, const int lda, const int nbase_x)
+    {
+        std::complex<double>* aux = new std::complex<double>[lda*n];
```

Comment:
This violates the RAII rule of cpp. Please use with caution.

Reply:
If the data in the entire CG process is changed to column-first in the near future, this function will be deprecated. So I did not optimize it here.